Remove the chevron icons from the “Finish setup” task list

adrianduffell opened the pull request:
Fixes #4896

This aligns with the update of the Store Management card in #4790 by removing the chevron icon from the item rows in this task list.

Now, only a "Dismiss" button renders in the right-side area of a row. This occurs if the task is optional and incomplete. Due to this a small refactor is made to remove the conditional operator when assigning `task.after`.

A test is added to assert the logic for rendering the "Dismiss" button.

### Screenshots

![Screen Shot 2020-09-09 at 1 48 10 pm](https://user-images.githubusercontent.com/9312929/92561162-42f3fa80-f2a6-11ea-95e7-9ead9f31d7d9.png)

### Detailed test instructions:

- Start on a site that hasn’t finished the setup tasks, and has the "Purchase & install extensions" task still to do.
- Confirm that chevron icons have been removed from the right-hand side of each incomplete task.
- Confirm that a "Dismiss" button is still present for the  "Purchase & install extensions" task.

### Changelog Note:

Tweak: The "Finish setup" task list no longer displays a chevron icon for incompleted tasks.

adrianduffell committed:
Remove the chevron icons from the “Finish setup” task list

This aligns with the update of the Store Management card in GH-4790 by removing the chevron icon from the item rows in this task list.

A small refactor was made to remove the conditional operator when assigning `task.after`. This is because we no longer need to render anything if the task is not dismissible.

adrianduffell committed:
Add a data-testid attribute to the TaskDashboard dismiss button for testing

This provides a reliable way to query the button in tests.

adrianduffell committed:
Add test for dismiss button in TaskDashboard component

samueljseay commented on client/task-list/test/index.js:
```diff
@@ -0,0 +1,72 @@
+	};
+} );
+
+describe( 'TaskDashboard', () => {
```
Great to see some testing 🎉

samueljseay commented on client/task-list/test/index.js:
```diff
@@ -0,0 +1,72 @@
+		render( <TaskDashboard query={ {} } /> );
+
+		// Wait for the task list to render.
+		expect( await screen.findByText( 'Finish setup' ) ).toBeDefined();
```
Not requesting a change but just an FYI that you mostly never need to query the whole screen for component tests. Rather you can capture the container when calling render and then query that.

e.g.

```
const {container} = render(<TaskDashboard query={{}} />);

expect( container.findByText('Finish setup')).toBeDefined();
```

adrianduffell replied:
Thanks! I'll make this change.

adrianduffell replied:
I am querying the container now in 985f8d3e39b9012db51fd65de74a05db76878f5b 😄

samueljseay commented on client/task-list/index.js:
```diff
@@ -298,9 +298,10 @@ class TaskDashboard extends Component {
 
-			if ( ! task.completed ) {
-				task.after = task.isDismissable ? (
+			if ( ! task.completed && task.isDismissable ) {
```
nice change 😄

samueljseay reviewed (approved):
Pre-approving this. I've tested the code locally and it works well.

adrianduffell committed:
Refactor element queries in test for performance

A small performance improvement is made by querying the rendered container instead of the whole screen.
